shell/pages/c/_cluster/fleet/graph/config.js

```diff
@@ -0,0 +1,277 @@
+        state:          bundle.state,
+        stateLabel:     bundle.stateDisplay,
+        stateColor:     bundleStateColor,
+        errorMsg:       bundle.stateDescription,
+        detailLocation: bundle.detailLocation,
+        children:       []
+      };
+
+      const bds = data.bundleDeployments.filter((bd) => bundle.id === `${ bd.metadata?.labels?.['fleet.cattle.io/bundle-namespace'] }/${ bd.metadata?.labels?.['fleet.cattle.io/bundle-name'] }`);
+
+      bds.forEach((bd) => {
+        const bdLowercaseState = bd.state ? bd.state.toLowerCase() : 'unknown';
+        const bdStateColor = STATES[bdLowercaseState]?.color;
+
+        const cluster = data.clustersList.find((cluster) => {
+          const clusterString = `${ cluster.namespace }-${ cluster.name }`;
+
+          return bd.id.includes(clusterString);
+        });
+
+        appChild.children.push({
+          id:                    bd.id,
+          type:                  bd.type,
+          matchingId:            `${ bd.type }-${ bd.id }`,
+          clusterLabel:          cluster ? cluster.namespacedName : undefined,
+          clusterDetailLocation: cluster ? cluster.detailLocation : undefined,
+          state:                 bd.state,
+          stateLabel:            bd.stateDisplay,
+          stateColor:            bdStateColor,
+          errorMsg:              bd.stateDescription,
+          detailLocation:        bd.detailLocation,
+        });
+      });
+
+      return appChild;
+    });
+
+    const appLowercaseState = data.state ? data.state.toLowerCase() : 'unknown';
+    const appStateColor = STATES[appLowercaseState].color;
+
+    return {
+      id:             data.id,
+      type:           data.type,
+      matchingId:     `${ data.type }-${ data.id }`,
+      state:          data.state,
+      stateLabel:     data.stateDisplay,
+      stateColor:     appStateColor,
+      errorMsg:       data.stateDescription,
+      detailLocation: data.detailLocation,
+      children:       bundles,
+      muteStatus:     true
+    };
+  },
+  /**
+   * Used to add relevant classes to each main node instance
+   */
+  extendNodeClass: ({ data }) => {
+    const classArray = [];
+
+    if (data?.type) {
+      const nodeType = data.type.replaceAll('fleet.cattle.io.', '');
+
+      classArray.push(nodeType);
+    }
+
+    return classArray;
+  },
+  /**
+   * Used to add the correct icon to each node
+   */
+  fetchNodeIcon: ({ data }) => {
+    let type = '';
+
+    switch (data?.type) {
+    case FLEET.GIT_REPO:
+      type = 'git';
+      break;
+    case FLEET.HELM_OP:
+      type = 'helm';
+      break;
+    case FLEET.BUNDLE:
+      if (data?.id.indexOf('helm') !== -1) {
+        type = 'helm';
+      }
+
+      type = 'bundle';
+      break;
+    case FLEET.BUNDLE_DEPLOYMENT:
+      type = 'node';
+      break;
+    }
+
+    return chartIcon(type);
+  },
+  /**
+     * Used to set node dimensions
+     */
+  nodeDimensions: ({ data }) => {
```

Comment:
Not a big deal, but to create a static version of all the properties with a indexation, it would help to identify better them, what do you think? Then later you could use a `switch` as well.